xdsl/interpreters/builtin.py

```diff
@@ -85,7 +88,7 @@ def dense_int_or_fp_elements_value(
         attr: Attribute,
         type_attr: builtin.MemRefType[Any],
     ) -> ShapedArray[Any]:
-        assert isinstance(attr, builtin.DenseIntOrFPElementsAttr)
+        assert isattr(attr, base(builtin.DenseIntOrFPElementsAttr[AnyDenseElement]))
```

Comment:
To expand on this thought, it's because `base` does the type parsing every time this is called, whereas Constr caches that, and is a tiny bit more resilient to API changes in the future